docs/releasing.md

```diff
@@ -1,96 +1,184 @@
 
-To generate new manifests and perform basic checks:
+    ```sh
+    $ git checkout relesae-3.0
+    $ git cherry-pick [COMMIT_IDS...]
+    ```
 
-    make prepare-release -B VERSION=v2.11.5
+2. Hope for few merge conflicts!
 
-Publish the images and local Git changes (disabling K3D as this is faster and more reliable for releases):
+3. Once done cherry-picking, push the release branch to ensure the branch can build and all tests pass.
 
-    make publish-release K3D=false VERSION=v2.11.5
-    
-Wait 1h to 2h.
+### 3. Prepare the Release
 
-* [ ] Check the images were pushed successfully.
-* [ ] Check the correct versions are printed.
-* [ ] Check the executor was correctly built.
+#### NOTE: Releasing for `v2`
 
+`v2` releases still depend on the previous repository name (`github.com/argoproj/argo`). To release for `v2`,
+make a local clone of the repository under the name `argo`:
+
+```shell
+$ pwd
+/Users/<user>/go/src/github.com/argoproj/argo-workflows
+$ cd ..
+$ cp -r argo-workflows argo
+$ cd argo
 ```
-docker run argoproj/argoexec:v2.11.5 version
-docker run argoproj/workflow-controller:v2.11.5 version
-docker run argoproj/argocli:v2.11.5 version
-```
 
-* [ ] Check the manifests contain the correct tags: https://raw.githubusercontent.com/argoproj/argo-workflows/v2.11.5/manifests/install.yaml
+Then follow all the normal steps. You should delete the `argo` folder once the release is done to avoid confusion and conflicts.
+
+#### Preparing the release
+
+1. Releasing requires a clean tree state, so back-up any untracked files in your Git directory.
+   
+   **Only once your files are backed up**, run:
+      ```shell
+       $ git clean -fdx  # WARNING: Will delete untracked files!
+      ```
+
+2. To generate new manifests and perform basic checks:
+   
+      ```shell
+      $ make prepare-release -B VERSION=v3.0.3
+      ```
+
+3. Once done, push the release branch and ensure the branch is green and all tests pass.
+
+      ```shell
+      $ git push
+      ```
+
+4. Publish the images and local Git changes (disabling K3D as this is faster and more reliable for releases):
+
+      ```shell
+      $ make publish-release K3D=false VERSION=v3.0.3
+      ```
+
+5. Wait 1h to 2h.
+
+### 4. Ensure the Release Succeeded
+
+1. Check the images were pushed successfully. Ensure the `GitTreeState` is `Clean`.
+   ```sh
+   $ docker run argoproj/argoexec:v3.0.3 version
+   $ docker run argoproj/workflow-controller:v3.0.3 version
+   $ docker run argoproj/argocli:v3.0.3 version
+   ```
+   
+1. Check the correct versions are printed. Ensure the `GitTreeState` is `Clean`.
+   ```sh
+   $ ./dist/argo-darwin-amd64 version
+   ```
 
-* [ ] Check the manifests apply: `kubectl -n argo apply -f https://raw.githubusercontent.com/argoproj/argo-workflows/v2.11.5/manifests/install.yaml`
+1. Check the manifests contain the correct tags (search for `v3.0.3`): https://raw.githubusercontent.com/argoproj/argo-workflows/v3.0.3/manifests/install.yaml
```

Comment:
Should this be a link? Looks like MkDocs won't render this into a link automatically.

![Screen Shot 2021-05-04 at 12 45 53 AM](https://user-images.githubusercontent.com/1311594/116961730-1ac06e00-ac72-11eb-9bb0-ee97b55c12d6.png)


```suggestion
1. Check the manifests contain the correct tags (search for `v3.0.3`): [https://raw.githubusercontent.com/argoproj/argo-workflows/v3.0.3/manifests/install.yaml](https://raw.githubusercontent.com/argoproj/argo-workflows/v3.0.3/manifests/install.yaml)
```